### DISCUSSION TITLE
This is fantastic

### DISCUSSION THREAD

**Discussion by Cod053** [PRProd/HA-Firemote#297] [General]
I just wanted to say that, probably like many other people, I had been searching for something like this for ages for my Nvidia Shield. This integration is excellent and super easy to use. I was struggling to get the apps into the official ADB or Android TV Remote integrations (still am), but this remote just made it happen like magic. So a massive thankyou to you for your effort - and your excellent documentation as well!

Next thing I need to figure out is how to expose the commands to Alexa / Google (and HA local voice control once the first two are replaced!) so that I can use voice commands to change the inputs (so, for example, being able to say "Alexa, change Shield input to Prime" and it actually works) or use the commands to insert into an automation (like setting up an automation to turn on TV, turn on AV, turn on Shield, and change Shield to the correct input). I currently have a Logitech Harmony Elite which does that, and I have the  Harmony integrated into HA and the Harmony activities exposed via Nabu Casa to Google and Alexa for voice control, but it's a bit of a pain, involving programming button presses, or using Flirc and button remapping, and eventually Logitech will turn off the servers anyway. So using HA to just run the automations, and expose them to voice assistants would be the goal.

**Comment by PRProd** [comment]
@Cod053 ~

THANK YOU FOR YOUR KIND WORDS!  They mean SO MUCH to me!  I'm glad you found my project, found that it was easy for you to use, AND I'm glad that my documentation has been helpful for you!

Your comments about my documentation and about exposing commands to your Alexa / Google / HA Voice control got me thinking, (I've heard these things from others as well) so I just took some time to add more clarity to the [How Does Firemote Send Commands to my Devices](https://github.com/PRProd/HA-Firemote/wiki#how-does-firemote-send-commands-to-my-devices) article in the [Firemote wiki](https://github.com/PRProd/HA-Firemote/wiki) which you might find useful.  

I've thought about innovators like you and wondered if it would be worthwhile to add a feature to Firemote that would spit out the Home Assistant commands that are invoked when a button is pressed.  I could be wrong about that idea... Would you find something like that to be helpful?  I think it has the potential to save you a couple of minutes of research when trying to replicate Firemote features for your own projects.

**Reply by Cod053** [reply]
You're most welcome. Credit should always be given where it's due 😊

Also, I should correct myself from calling this an integration, particularly since you clearly pointed out it's actually a card, built off of the ADB. Regardless, it's still excellent work.

I think the idea of being able to get a note of the invoked commands generated would definitely be pretty useful. 

I've actually got an Unfolded Circle Remote Two to set up, and an Allonis Smart Remote and myServer due in the next week (I know, I should have just picked one, but I love new shiny tech 🤣), both of which have HA integrations, so that might be quite helpful when I'm setting those up.

**Reply by Cod053** [reply]
Also, it should be noted that I can actually change inputs by choice via Google Home at the moment, but it's not local, and it's subject to breaking because, well, Google. Something running on HA and either exposed locally to HA Assist or through Nabu Casa to Home or Alexa is always going to be better.

**Reply by PRProd** [reply]
I'm totally with you on the idea of keeping things local!  Always fast, always reliable, and always free!  :smile:

I looked into those remote controls that you mentioned and :exploding_head: wow!  Soooo awesome looking!  I'd be curious to know which one is your favorite after using them all for a couple of months.

**Reply by PRProd** [reply]
@Cod053 ~ I mentioned this discussion thread in a [Reddit post I made about the recent release of Roku support](https://www.reddit.com/r/homeassistant/comments/19eol16/firemote_keeps_getting_better/?utm_source=share&utm_medium=web2x&context=3), and one user pointed me to this project: https://github.com/CoretechR/OMOTE .  If you haven't seen this project before, you might be interested.

I didn't look too deeply into it yet, but at first glance, it earned a star from me!  It looks really cool!
